pkg/boot/universalpayload/utilities_test.go

```diff
@@ -454,3 +457,358 @@ func TestConstructSMBIOS3Node(t *testing.T) {
+			"0x0000000800000000 0x0000000800EFFFFF 0x0000000000140204\n",
+			"0x0000000000000000 0x0000000000000000 0x0000000000000000\n",
+		},
+		{
+			// Content for "0000:15:02.0"
+			"0x00000000B8000000 0x00000000B87FFFFF 0x0000000000040200\n",
+			"0x0000000000000000 0x0000000000000000 0x0000000000000000\n",
+		},
+		{
+			// Content for "0000:C8:00.0"
+			"0x00000000B8800000 0x00000000B8FFFFFF 0x0000000000040200\n",
+			"0x0000000000000000 0x0000000000000000 0x0000000000000000\n",
+			"0x000000000000E000 0x000000000000E03F 0x0000000000040101\n",
+			"0x0000000000000000 0x0000000000000000 0x0000000000000000\n",
+			"0x0000000800F00000 0x0000000800FFFFFF 0x0000000000140204\n",
+		},
+		{
+			// Content for "0000:E1:00.0"
+			"0x00000000B0000000 0x00000000B0FFFFFF 0x0000000000040200\n",
+			"0x0000000801000000 0x0000000801FFFFFF 0x0000000000140204\n",
+		},
+		{
+			// Content for "0000:FF:00.0"
+			"0x00000000A8000000 0x00000000AFFFFFFF 0x0000000000040200\n",
+			"0x0000000000000000 0x0000000000000000 0x0000000000000000\n",
+		},
+		{
+			// Content for "0002:00:00.0"
+			"0x0000000810000000 0x000000081000FFFF 0x0000000000140204\n",
+			"0x0000000000000000 0x0000000000000000 0x0000000000000000\n",
+		},
+		{
+			// Content for "0002:00:02.0"
+			"0x00000000A0070000 0x00000000A07FFFFF 0x0000000000040200\n",
+			"0x000000000000B000 0x000000000000BFFF 0x0000000000040101\n",
+			"0x00000000000C0000 0x00000000000DFFFF 0x0000000000000212\n",
+			"0x0000000000000000 0x0000000000000000 0x0000000000000000\n",
+			"0x0000000810010000 0x000000081001FFFF 0x0000000000140204\n",
+			"0x0000000000000000 0x0000000000000000 0x0000000000000000\n",
+		},
+		{
+			// Content for "0002:1f:00.0"
+			"0x00000000A0040000 0x00000000A004FFFF 0x0000000000040200\n",
+			"0x00000000A0060000 0x00000000A006FFFF 0x0000000000040200\n",
+		},
+		{
+			// Content for "0002:af:00.0"
+			"0x00000000A0030000 0x00000000A003FFFF 0x0000000000040200\n",
+			"0x0000000810020000 0x00000008103FFFFF 0x0000000000140204\n",
+			"0x0000000000000000 0x0000000000000000 0x0000000000000000\n",
+		},
+		{
+			// Content for "0002:ff:00.0"
+			"0x00000000A0000000 0x00000000A000FFFF 0x0000000000040200\n",
+			"0x000000000000A000 0x000000000000A03F 0x0000000000040101\n",
+			"0x0000000000000000 0x0000000000000000 0x0000000000000000\n",
+		},
+	}
+
+	expectedResourceRegion := []ResourceRegions{
+		{
+			Mmio64Base:  0x0000_0008_0000_0000,
+			Mmio64Limit: 0x0000_0000_0100_0000,
+			Mmio32Base:  0x0000_0000_B800_0000,
+			Mmio32Limit: 0x0000_0000_2700_0000,
+			IoPortBase:  0x0000_0000_0000_E000,
+			IoPortLimit: 0x0000_0000_0000_2000,
+		},
+		{
+			Mmio64Base:  0x0000_0008_1000_0000,
+			Mmio64Limit: 0x0000_0000_0040_0000,
+			Mmio32Base:  0x0000_0000_A000_0000,
+			Mmio32Limit: 0x0000_0000_0080_0000,
+			IoPortBase:  0x0000_0000_0000_A000,
+			IoPortLimit: 0x0000_0000_0000_2000,
+		},
+	}
+
+	// Create temp folder for testing: /tmp/PciRootBridgeInfoTest
+	tmpDir := os.TempDir()
```

Comment:
You can use t.TempDir and in this case you don't os.Remove

Reply:
Thanks, let me try it later.